client/lib/route-modal/use-route-modal.tsx

```diff
@@ -0,0 +1,47 @@
+import { addQueryArgs, removeQueryArgs } from '@wordpress/url';
+import page from 'page';
+import { useSelector } from 'react-redux';
+import getCurrentQueryArguments from 'calypso/state/selectors/get-current-query-arguments';
+import getPreviousRoute from 'calypso/state/selectors/get-previous-route.js';
+
+interface ReturnValues {
```

Comment:
As we're not exporting this, can this be `type` instead of `interface`?

Reply:
I have no preference so let's keep it `interface`  as the [official document](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#differences-between-type-aliases-and-interfaces) also said

> If you would like a heuristic, use interface until you need to use features from type.

Reply:
Oh wow, TIL about this! Sure sure